org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradlePreferenceChangeListener.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2016-2020 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.managers;
+
+import java.util.Objects;
+
+import org.eclipse.buildship.core.GradleDistribution;
+import org.eclipse.buildship.core.WrapperGradleDistribution;
+import org.eclipse.buildship.core.internal.CorePlugin;
+import org.eclipse.buildship.core.internal.configuration.ProjectConfiguration;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.ProjectUtils;
+import org.eclipse.jdt.ls.core.internal.preferences.IPreferencesChangeListener;
+import org.eclipse.jdt.ls.core.internal.preferences.Preferences;
+
+/**
+ * @author snjeza
+ *
+ */
+public class GradlePreferenceChangeListener implements IPreferencesChangeListener {
+	@Override
+	public void preferencesChange(Preferences oldPreferences, Preferences newPreferences) {
+		ProjectsManager projectsManager = JavaLanguageServerPlugin.getProjectsManager();
+		if (projectsManager != null) {
+			if ((!Objects.equals(oldPreferences.getSha256Allowed(), newPreferences.getSha256Allowed()) || !Objects.equals(oldPreferences.getSha256Disallowed(), newPreferences.getSha256Disallowed()))) {
+				for (IProject project : ProjectUtils.getGradleProjects()) {
+					ProjectConfiguration configuration = CorePlugin.configurationManager().loadProjectConfiguration(project);
+					GradleDistribution distribution = configuration.getBuildConfiguration().getGradleDistribution();
+					if (distribution instanceof WrapperGradleDistribution) {
+						projectsManager.updateProject(project, true);
```

Comment:
I'm curious why need update project when user changes checksums user setting? 

I installed the latest 0.62.3 fix, and have a play on [spring-boot](https://github.com/spring-projects/spring-boot) project, when i changed checksums, it will trigger lots of jobs. Is there any way to optimize it?
![updateProject](https://user-images.githubusercontent.com/14052197/83849476-1332d500-a742-11ea-9ffd-98b7369ffcf0.gif)

Reply:
Fixed.